netplan/cli/commands/apply.py

```diff
@@ -290,17 +295,25 @@ def command_apply(self, run_generate=True, sync=False, exit_on_error=True, state
                 utils.ip_addr_flush(iface)
             # clear NM state, especially the [device].managed=true config, as that might have been
             # re-set via an udev rule setting "NM_UNMANAGED=1"
-            subprocess.check_call(['rm', '-rf', '/run/NetworkManager/devices'])
+            subprocess.check_call(['rm', '-rf', '/run/NetworkManager/devices'], env=env)
```

Comment:
alternative approach is to use `shutil.rmtree` but it's not exactly equivalent so take it with a grain of salt :)

Reply:
ACK. This is the cleaner approach and also being used elsewhere in the codebase.

Reply:
see #338